### PR TITLE
feat: delete tmp files

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-237

### CODE REVIEW

File: helpers/class.ItemResolver.php [hunk 1]
```diff
@@ -36,15 +36,19 @@
...
     use OntologyAwareTrait;
-    
+
     /**
      * @var Service
      */
     private $service;
-    
+
+    /** @var array */
```
Comment (boajer) [inline helpers/class.ItemResolver.php:45]:
```suggestion
    /** @var string[] */
```

File: helpers/class.ItemResolver.php [hunk 2]
```diff
@@ -76,7 +80,18 @@ public function resolve($url)
...
+
+        $this->tmpFiles[] = $tmpfile;
+
         return $tmpfile;
     }
+
+    public function __destruct()
+    {
+        foreach ($this->tmpFiles as $tmpFile) {
+            if (file_exists($tmpFile) && is_writable($tmpFile)) {
```
Comment (boajer) [inline helpers/class.ItemResolver.php:92]:
```suggestion
            if (is_writable($tmpFile)) {
```
As https://www.php.net/manual/en/function.is-writable.php `Returns true if the filename exists and is writable`

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Review by voknil — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by voknil — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**